src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ParameterBinder.cs

```diff
@@ -100,15 +100,52 @@ public class ParameterBinder
                 CacheToken = parameter,
             });
 
-            var modelBindingContext = DefaultModelBindingContext.CreateBindingContext(
+            return BindModelAsync(
                 actionContext,
+                binder,
                 valueProvider,
+                parameter,
                 metadata,
+                value);
+        }
+
+        public virtual async Task<ModelBindingResult> BindModelAsync(
+            ActionContext actionContext,
+            IModelBinder modelBinder,
+            IValueProvider valueProvider,
+            ParameterDescriptor parameter,
+            ModelMetadata modelMetadata,
```

Comment:
is `modelMetadata` allowed to be null? It's also called `metadata` elsewhere in this file, choose one